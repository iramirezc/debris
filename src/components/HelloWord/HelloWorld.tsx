import React from "react"
import { log } from "utils"

export const HelloWorld: React.FC<{ message?: string }> = ({
  message = "Hello World!"
}) => {
  log("Hello World!")

  return <h1>{message}</h1>
}
