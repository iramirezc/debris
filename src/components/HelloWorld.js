import React from "react";

export const HelloWorld = () => {
  console.log("Hello World!");

  return <h1>Hello World!</h1>;
};
