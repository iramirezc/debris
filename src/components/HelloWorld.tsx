import React from "react"
import { log } from "utils/index"

export const HelloWorld: React.FC<{ message?: string }> = ({
  message = "Hello World!",
}) => {
  log("Hello World!")

  return <h1>{message}</h1>
}
