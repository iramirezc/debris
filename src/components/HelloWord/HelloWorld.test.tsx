import React from "react"
import { render } from "@testing-library/react"
import { HelloWorld } from "./HelloWorld"

jest.mock("utils", () => ({
  log: jest.fn()
}))

describe("<HelloWorld/>", () => {
  beforeAll(() => {
    jest.spyOn(console, "log").mockImplementation()

    global.fetch = jest.fn().mockResolvedValue({
      json: jest.fn().mockResolvedValue({ status: "ok" })
    })
  })

  test("renders with the default 'Hello World!' message", () => {
    const { container } = render(<HelloWorld />)

    expect(container.firstChild).toMatchSnapshot()
  })

  test("renders with the provided message", () => {
    const { getByRole } = render(<HelloWorld message="Auf Wiedersehen" />)

    expect(getByRole("heading")).toHaveTextContent("Auf Wiedersehen")
  })
})
