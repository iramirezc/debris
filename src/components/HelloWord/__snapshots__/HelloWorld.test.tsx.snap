// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWorld/> renders with the default 'Hello World!' message 1`] = `
<h1
  class="sc-bcXHqe gQSvAQ"
>
  Hello World!
</h1>
`;
